crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -30,6 +30,18 @@ def node_status
         }
       end
 
+      def upgrade_mode
+        ::Crowbar::UpgradeStatus.new.upgrade_mode
+      end
+
+      def upgrade_mode=(mode)
+        unless ["normal", "non_disruptive"].include? mode
+          raise ::Crowbar::Error::SaveUpgradeModeError, "Invalid upgrade mode #{mode}."
```

Comment:
Hm, I actually added that to crowbar-client already (the PR for that is will follow soon). Buy yeah, I can add it here as well.